### DISCUSSION TITLE
CORS header ‘Access-Control-Allow-Origin’ missing

### DISCUSSION THREAD

**Discussion by eliascotto** [fastapi/fastapi#9027] [Questions]
## Issue
`Access-Control-Allow-Origin` is missing from the response header with the code below.

### Example

```Python
from starlette.middleware import Middleware
from starlette.middleware.cors import CORSMiddleware
from config import ENV_NAME

if ENV_NAME == "development":
    origins = ["*"]
else:
    origins = [
        "https://x.vercel.app/",
        "http://x.vercel.app/"
    ]

middlewares = [
    Middleware(
        CORSMiddleware,
        allow_origins=origins,
        allow_credentials=True,
        allow_methods=["*"],
        allow_headers=["*"]
    )
]

app = FastAPI(middleware=middlewares)
```

### Description
Happens in production version, while using `docs#/`

- Tested with `development` env locally and works:
```
access-control-allow-origin: http://127.0.0.1:8000
```

### Environment

Cloud: cloud.digitalocean.com

```txt
fastapi==0.62.0
gunicorn==20.0.4
starlette==0.13.6
text-unidecode==1.3
uvicorn==0.12.3
```

**Comment by falkben** [comment]
I'd need more information in order to understand what the problem is.

What is the request? Does it have origin specified? Are you using a reverse proxy? How is the request being generated? 

Please give as much example code as you can.

**Comment by yuriescl** [comment]
I have a similar issue, but it's more like a question.
I'm setting the CORS middleware in my Fastapi server, but the response has no CORS headers.
`app.py`:
```
app = FastAPI()
app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)
```

curl response from `localhost:8000` (which is the fastapi server):
```
*   Trying 127.0.0.1:8000...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8000 (#0)
> GET / HTTP/1.1
> Host: localhost:8000
> User-Agent: curl/7.68.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 422 Unprocessable Entity
< date: Fri, 26 Nov 2021 13:01:30 GMT
< server: uvicorn
< content-length: 176
< content-type: application/json
<
* Connection #0 to host localhost left intact
```

Ignore the 422 Unprocessable Entity.
Why are there no CORS headers in the response?

**Comment by yuriescl** [comment]
> Why are there no CORS headers in the response?

Because the request is missing the `Origin` header, which is required in order for the CORS middleware to add the CORS headers in the response, as mentioned [here](https://fastapi.tiangolo.com/tutorial/cors/#simple-requests).

**Comment by AlexTraveylan** [comment]
No idea.

I tried to be explicit:

![image](https://github.com/user-attachments/assets/ed32030d-e2b0-42a7-ab6b-7b028b0c70fd)

I tried to setup a vercel.json

![image](https://github.com/user-attachments/assets/0bc914e1-604c-4d44-ac66-fe62aca217e6)

Everything cause CORS problems 

![image](https://github.com/user-attachments/assets/fbd1a10f-9f48-42ca-a6e5-ead577663929)

I hope someone will find a solution
